.gitignore

```diff
@@ -129,4 +129,7 @@ dmypy.json
 .pyre/
 
 # Editors
-.vscode/
\ No newline at end of file
+.vscode/
+
+# dynamically generated docs
+docs/source/component_docs.rst
```

Comment:
Put this under `# Sphinx documentation`?

Reply:
The first 90% of the file is a direct match for the default .gitignore for Python projects that GitHub provides. I would rather keep that separate from the custom entries we put in the file. I'll update the comments in the .gitignore file to reflect this and also swap the order of the ignore rules so that our custom ones are first.

I think this will make it easier to keep track of what we are ignoring rather than mixing the generic rules (some of which we don't technically need) with our custom entries.

Reply:
I've actually put this back the way it was now - as the custom rules section should really come last so that tools that append to the .gitignore file are effectively adding it to the section clearly demarked as part of our custom rules. Hope that makes sense.